nodes/delete_node.js

```diff
@@ -1,30 +1,87 @@
 module.exports = function (RED) {
+  'use strict';
+
   const { WazoApiClient } = require('@wazo/sdk');
+  
+  function setProperty(node, msg, name, type, value) {
+      if (type === 'msg') {
+          msg[name] = value;
+      } else if (type === 'flow') {
+          node.context().flow.set(name, value);
+      } else if (type === 'global') {
+          node.context().global.set(name, value);
+      }
+  }
+
+  function sendErrorMessage(node, text) {
+      node.status({
+          fill: 'red',
+          shape: 'ring',
+          text: (typeof text === "object") ? JSON.stringify(text) : text
+      });
+  }
+  
+  function setStatus(node, text, color, shape) {
+    if (!text && !color && !shape) {
+      node.status({});
+    } else {
+      node.status({
+        fill: color || 'blue',
+        shape: shape || 'ring',
+        text: text
+      });
+    }
+  }
+  
+  function checkType(node, parameter, type) {
+    if (typeof parameter !== type) {
+      const text = RED._('delete_node.errors.falsetype') + type;
+      sendErrorMessage(node, text);
+      throw(text);
+    }
+  }
+
 
   function delete_node(n) {
-    RED.nodes.getNode(this, n);
+    RED.nodes.createNode(this, n);
     conn = RED.nodes.getNode(n.server);
     this.client = conn.client.application;
+    
+    let node = this;
+    setStatus(node);
+
+    node.callnode = n.callnode;
+    node.callnodeType = n.callnodeType;
 
-    var node = this;
+    node.on('input', async (msg, send, done) => {
```

Comment:
You are using the new method with node-red. We need to assume the old node-red installation won't work. But we need to change that on all nodes.

Reply:
I will do that, when I rework the other nodes.